core/builtin.py

```diff
@@ -483,9 +483,24 @@ def Cd(argv, mem):
   return 0
 
 
-def PrintDirStack(dir_stack):
-  if dir_stack:
-    print(' '.join(dir_stack))
+WITH_PREFIX = 1
+WITHOUT_PREFIX = 2
+SINGLE_LINE = 3
+def PrintDirStack(dir_stack, mode = 3):
```

Comment:
Remove the default 3 and just pass SINGLE_LINE in the other few places.

Python has some pitfalls with regard to mixing default values and passing positional args (as opposed to named args).